src/App.jsx

```diff
@@ -14,33 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [sortField, setSortField] = useState('');
+  const [showReset, setShowReset] = useState(false);
+  const [reverse, setReverse] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const toSort = field => {
+    let sortedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    switch (field) {
+      case 'Sort alphabetically':
+        sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case 'Sort by length':
+        sortedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      case 'Reverse':
+        sortedGoods.reverse();
+        setReverse(!reverse);
+        break;
+      case 'Reset':
+        sortedGoods = [...goodsFromServer];
+        break;
+      default:
+        break;
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    setGoods(sortedGoods);
+  };
```

Comment:
move sorting function to a helper outside the App component. Here is how it should looks like

* if you are familiar with `enum` you can move `length` and `alphabetically` strings there
```
function getSortGoods(goods, sortField, isReversed) {
  const preparedGoods = [...goods];

  if (sortField) {
    preparedGoods.sort((good1, good2) => {
      switch (sortField) {
        case 'length':
          return good1.localeCompare(good2);
        case 'alphabetically':
          return good1.length - good2.length;
        default:
          return 0;
      }
    });
  }

  if (isReversed) {
    preparedGoods.reverse();
  }

  return preparedGoods;
}
```